github-actions/action.yml

```
@@ -0,0 +1,49 @@
+# Copyright 2020 SUSE
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: GitHub Actions Doc Generator
+description: Generates Markdown docs for GitHub Actions.
+inputs:
+  exclude:
+    description: A list of paths (one per line) to exclude from the generation.
+    required: false
+    default: |-
+      third-party/
+outputs: {}
+runs:
+  using: composite
+  steps:
+  - run: |-
+      if [ -z '${{ github.head_ref }}' ]; then
+        >&2 echo "Not a pull request..."
+        exit 1
+      fi
+    shell: bash
+  - run: ${{ github.action_path }}/generate.rb
+    shell: bash
+    env:
+      EXCLUDE: ${{ inputs.exclude }}
+  - run: |-
+      set -o errexit -o nounset -o pipefail
+
+      files=($(git ls-files --deleted --modified --others **/*README.md))
```

`#include <shellcheck>`
```suggestion
      mapfile -d '' -t < <(git ls-files -z --deleted --modified --others **/*README.md)
```